pkg/datapath/linux/devices_controller_test.go

```diff
@@ -244,8 +231,13 @@ func TestDevicesController(t *testing.T) {
 		},
 	}
 
-	withFreshNetNS(t, func(ns netns.NsHandle) {
-
+	newNet, err := netns.New()
```

Comment:
Tip: create a new file `pkg/testutils/netns.go` with:

```
// SPDX-License-Identifier: Apache-2.0
// Copyright Authors of Cilium

package testutils

import (
	"testing"

	"github.com/cilium/cilium/pkg/netns"
)

// NewNetNS returns a new network namespace.
func NewNetNS(tb testing.TB) *netns.NetNSHandle {
	tb.Helper()

	ns, err := netns.New()
	if err != nil {
		tb.Fatal(err)
	}
	tb.Cleanup(func() {
		ns.Close()
	})

	return &ns
}
```

and simply `ns := testutils.NewNetNS(t)` eveywhere.

Reply:
Yep, better next step after removing most/all of the local helper funcs.

Reply:
though this is somewhat incompatible with making `NetNSHandle` private.